src/data/data_fetcher.py

```diff
@@ -106,8 +106,10 @@ def _make_api_url(
 def _handle_two_year_transaction_period(
     two_year_transaction_period: Union[int, str]
 ) -> str:
-    if two_year_transaction_period.isnumeric():
-        if (int(two_year_transaction_period) % 2) == 0:
+    if str(two_year_transaction_period).isnumeric():
+        if int(two_year_transaction_period) % 2 != 0:
+            two_year_transaction_period = int(two_year_transaction_period) + 1
+            two_year_transaction_period = str(two_year_transaction_period)
             if (
```

Comment:
Taking a step back, the fact that we have `print("Invalid input, defaulting to 2020.")` in multiple locations should set your spidey senses a'tingling that there's a cleaner way to write this.

Few leading questions for you:

1. Does this line need to be intented?
2. Do we *need* `else` statements?
3. Can we simplify our flow control by:
    * Inverting our check (year **in** the range vs year **out of** range)
    * Can we make good use of a mid-function `return` statement?

Consider the following pseudocode

```
if is_numeric
    if is_odd_year
       make_it_appropriate_even_year
    if in_bounds
       return str(value)
error_msg
return '2020'
```

Any records you pass in within the range won't ever make it to the `error_msg` line.

But don't take my word for it-- we wrote unit tests precisely so we can tinker on this function and be confident we're on the right track :)